### ISSUE TITLE
Allow pass name paramter in BaseMessageStringPromptTemplate

### ISSUE THREAD

**Issue by yueqing-wolai** [langchain-ai/langchainjs#735]
Based on OPENAI Cook book example, pass example_user and example_assisatant in message name is a way of few shot example, link is here [](https://github.com/openai/openai-cookbook/blob/main/examples/How_to_format_inputs_to_ChatGPT_models.ipynb).
response = openai.ChatCompletion.create(
    model=MODEL,
    messages=[
        {"role": "system", "content": "You are a helpful, pattern-following assistant that translates corporate jargon into plain English."},
        {"role": "system", "name":"example_user", "content": "New synergies will help drive top-line growth."},
        {"role": "system", "name": "example_assistant", "content": "Things working well together will increase revenue."},
        {"role": "system", "name":"example_user", "content": "Let's circle back when we have more bandwidth to touch base on opportunities for increased leverage."},
        {"role": "system", "name": "example_assistant", "content": "Let's talk later when we're less busy about how to do better."},
        {"role": "user", "content": "This late pivot means we don't have time to boil the ocean for the client deliverable."},
    ],
    temperature=0,
)
Will you consider this feature in the future?

**Comment by arronKler** [comment]
few-shot example do not need this kind of things. Just input it as normal Human input or system role input is enough. like:
```
here is some kind of examples for you:
user: New synergies will help drive top-line growth.
assistant: Things working well together will increase revenue.
user: Let's circle back when we have more bandwidth to touch base on opportunities for increased leverage.
assistant: Let's talk later when we're less busy about how to do better.
```
LLM just eat the whole input, the json data structure is just a program way, but eventually it will combine together and send to LLM

**Comment by jonaslalin** [comment]
TLDR: This feature would be great to have in LangChain.

To my understanding,

>The ChatGPT model (chat-davinci-003-alpha) is a language model designed for conversational interfaces, and thus operates slightly differently from previous GPT models such as text-davinci-002 and text-davinci-003.
>
>Previous models were text-in and text-out (i.e. they accepted a prompt string and returned a completion to append to the prompt). The ChatGPT model (GPT-3.5A), however, is conversation-in and message-out (i.e. it expects a prompt string that is formatted in a specific chat-like transcript format, and returns a completion that represents a model-written message in the chat).

Differences:
<img width="643" alt="Screenshot 2023-05-08 at 22 02 07" src="https://user-images.githubusercontent.com/23449085/236922261-33ebdbea-4114-4c9b-ab83-82c36724eaad.png">

Which means that few-shot learning with ChatGPT models looks closer to
```
<|im_start|>system
Translate from English to French
<|im_end|>
<|im_start|>system name=example_user
How are you?
<|im_end|>
<|im_start|>system name=example_assistant
Comment allez-vous?
<|im_end|>
<|im_start|>user
{{user input here}}<|im_end|>
```

Depending on how future models have been trained, passing examples as part of the system's input could make a big difference (e.g., providing protection against prompt injection).

Source: [ChatGPT API: Prompt formatting](https://docs.google.com/document/d/1mYBAIilR8IcIfzvIfrsayAU_XJJ-w5Oi6zYY53g0LFs/edit)

Also, see discussion in https://news.ycombinator.com/item?id=34988748.
